### PR TITLE
v1.2.0

### PR DESCRIPTION
**Pull request by kaero**
* fixed `Terror.isTerror() → false`, was `undefined`
* tests moved from jasmine and expectations to mocha and assertions (to keep it consistent with other nodules)
* fixed JSCS rules
* avoid usage of Function ctor to create derived constructor to avoid runtime errors in the case of error names which is not a valid JS identifiers. Also it looks like an open door for security issues and crazy hacks.

Published as `terror@1.2.0-rc2`

Only my changes: https://github.com/nodules/terror/compare/508bea3...v1.2

\cc @narqo @B-Vladi

### CODE REVIEW

**Comment by kaero** [comment]
@narqo I think we can test it in the wild and release if no issues were found.

File: benchmark/package.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "name" : "terror-benchmark",
+    "version" : "1.2.0",
+    "dependencies": {
+      "terror" : "git://github.com/nodules/terror.git#1.1.0",
```
Comment (narqo) [inline benchmark/package.json]: why git here?

File: test/terror.js [hunk 2]
```diff
@@ -1,337 +1,615 @@
...
+        var logger = function () {};
+
+        Terror.setLogger(logger);
+
+        assert.strictEqual(terror.logger, logger);
+
+        Terror.setLogger(defaultLogger);
+    });
+
+    describe('.stackTraceLimit()', function () {
```
Comment (B-Vladi) [inline test/terror.js]: This is not a function

Comment (kaero) [inline test/terror.js]: oh, thanks

**Comment by kaero** [comment]
Minor issue: if nested error is not an instance of Terror, then `undefined` prepends its error message in the full stack:
```
>>>>>     undefined SyntaxError: Unexpected token var
```

**Comment by kaero** [comment]
:up:

* fixed `undefined` which prepends messages of non-terror errors in the full stack
* different ways of calling the `Object#hasOwnProperty` reduced to `has.call(..)`

**Comment by kaero** [comment]
I've used 1.2.0-rc2 in the dev for app, which uses `1.1.0` and everything looks fine, except `undefined` in the messages, which i have fixed in 6be8ba3

**Comment by narqo** [comment]
looks good

**Comment by B-Vladi** [comment]
all right

**Pull request merged**